File: Module 0 - Programming Fundamentals/Datatypes/BasicDataTypes/Datatypes.cpp
/*  Data Types
      │
      ├── Integer - Stores whole numbers.
      ├── Float / Double - Stores decimal numbers.
      ├── Character - Stores a single character.
      ├── String - Stores a sequence of characters.
      └── Boolean - Stores either True or False */

      # include <iostream>
      using namespace std;

        int main() {
            int var1 = 10;
            float var2 = 10.5;
            char var3 = 'A';
            string var4 = "Hello, World!";
            bool var5 = true;
    
            cout << "Integer: " << var1 << endl;
            cout << "Float: " << var2 << endl;
            cout << "Character: " << var3 << endl;
            cout << "String: " << var4 << endl;
            cout << "Boolean: " << var5 << endl;
    
            return 0;
        }

        // C++ is a statically typed language,The variable type cannot change at runtime. Once a variable is declared with a specific type, it cannot be assigned a value of a different type without explicit casting.

File: Module 0 - Programming Fundamentals/Datatypes/BasicDataTypes/Datatypes.py
""" Data Types
      │
      ├── Integer - Stores whole numbers.
      ├── Float / Double - Stores decimal numbers.
      ├── Character - Stores a single character.
      ├── String - Stores a sequence of characters.
      └── Boolean - Stores either True or False """

# python treats characters as strings, so there is no separate character data type in Python.

var1 = 10
var2 = 10.5
var3 = 'A'
var4 = "Hello"
var5 = True

print("Integer:", var1)
print("Float:", var2)
print("Character:", var3)
print("String:", var4)
print("Boolean:", var5)

# python is Dynamic Typing Language, so we can change the type of variable at runtime

var1 = "Hello"  # Now var1 is a string
print("Var1 is Updated to String:", var1)